meta-xt-control-domain/recipes-guest/doma/u-boot-android-bazel.bb

```diff
@@ -0,0 +1,74 @@
+inherit base
+
+# PARAMETERS
+
+UBOOT_REPO_GIT_URL="github.com/xen-troops/android_u-boot_manifest"
+UBOOT_REPO_GIT_URL_DOTTED="github.com.xen-troops.android_u-boot_manifest"
+UBOOT_REPO_DOWNLOAD_PROTOCOL="https"
+UBOOT_REPO_GIT_BRANCH="xenvm-trout-main"
+UBOOT_REPO_MANIFEST="default.xml"
+
+# IMPLEMENTATION
+
+UBOOT_DOWNLOAD_REPO_DIR="${DL_DIR}/repo/${UBOOT_REPO_GIT_URL_DOTTED}/${UBOOT_REPO_MANIFEST}"
+UBOOT_LOCAL_REPO_RELATIVE_PATH="repo/${UBOOT_REPO_GIT_URL_DOTTED}/${UBOOT_REPO_MANIFEST}"
+UBOOT_LOCAL_REPO_DIR="${WORKDIR}/${UBOOT_LOCAL_REPO_RELATIVE_PATH}"
+UBOOT_BUILD_DIR="${B}/${UBOOT_REPO_GIT_URL_DOTTED}/${UBOOT_REPO_MANIFEST}/${U_BOOT_BUILD_TARGET}"
+
+SRC_URI="\
+    repo://${UBOOT_REPO_GIT_URL};protocol=${UBOOT_REPO_DOWNLOAD_PROTOCOL};branch=${UBOOT_REPO_GIT_BRANCH};manifest=${UBOOT_REPO_MANIFEST}  \
+"
+
+LICENSE="GPLv2+"
+LIC_FILES_CHKSUM="file://${WORKDIR}/${UBOOT_LOCAL_REPO_RELATIVE_PATH}/repo/u-boot/Licenses/README;md5=2ca5f2c35c8cc335f0a19756634782f1"
+
+FILES:${PN} = " \
+    ${libdir}/xen/boot/u-boot-doma \
+"
+
+U_BOOT_BUILD_TARGET="xen-guest-android-virtio_aarch64"
+
+do_unpack() {
+    if ! [ -d ${UBOOT_LOCAL_REPO_DIR} ]; then
+        mkdir -p ${UBOOT_LOCAL_REPO_DIR};
+        cp -r ${UBOOT_DOWNLOAD_REPO_DIR}/* ${UBOOT_LOCAL_REPO_DIR}/;
+    fi
+}
+
+python do_clean:prepend() {
+    import subprocess
+
+    UBOOT_LOCAL_REPO_DIR = d.getVar('UBOOT_LOCAL_REPO_DIR')
+
+    bash_command = f"if [ -d {UBOOT_LOCAL_REPO_DIR} ]; then " \
+    f"cd {UBOOT_LOCAL_REPO_DIR}/repo; " \
```

Comment:
Do you really need to define a python script that runs a bash script?

Reply:
Hi @lorc, in this case, I'm inheriting from base.bbclass.

base.bbclass inherits from utility-tasks.bbclass:
https://github.com/yoctoproject/poky/blob/kirkstone/meta/classes/base.bbclass#L9

And utility-tasks.bbclass implements "do_clean" action as a Python function:
https://github.com/yoctoproject/poky/blob/a43fa36614977745fcbd6441d1c3c20321a8201d/meta/classes-global/utility-tasks.bbclass#L31

It is not possible to mix bash and python formats. If the initial action is implemented as a Python script, then I should use Python for prepending.

I've actually started from having this part as a bash script, but I got the error during the build, which was coming from an attempt to execute by bash script as Python code.

Reply:
Hi @lorc,

```
1. Could you please align your recipe with the Yocto style guide? (https://www.openembedded.org/wiki/Styleguide)

2. Please split your commit into two. Remember the "atomic commits" rule.

3. "Origin" and "Original" are two distinct words with different meanings.
```

Points #2 & #3 are fully done.

Point #1 - done, except for adding HOMEPAGE and BUGTRACKER parameters. I'm not sure whether they are needed. And even if needed, which values to place there - the xen-troops' Github, or Google u-boot's location?